Compare-NET-Objects/TypeComparers/ListComparer.cs

```diff
@@ -64,8 +65,12 @@ public override void CompareType(CompareParms parms)
 
                 bool countsDifferent = ListsHaveDifferentCounts(parms);
 
-                if (parms.Config.IgnoreCollectionOrder && !ChildIsListOrDictionary(parms))
+                // If items is collections, need to use default compare logic, not ignore order logic.
+                // We cannot ignore order for nested collections because we will get an reflection exception.
+                // May be need to display some warning or write about this behavior in documentation.
+                if (parms.Config.IgnoreCollectionOrder && !ChildShouldBeComparedWithoutOrder(parms))
                 {
+                    // TODO: allow IndexerComparer to works with types (now it works only with properties).

```

Comment:
I decided to extend original check. Full support of collection with nested collection is not that simple as I thing. I suggest this feature for the future releases :)